Extensions/super_markdown_editor.js

```diff
@@ -0,0 +1,262 @@
+//* TITLE Super Markdown Editor **//
+//* VERSION 1.0 REV A **//
+//* DESCRIPTION Add live previews and other functionality to the Markdown post editor **//
+//* DEVELOPER new-xkit **//
+//* FRAME false **//
+//* BETA false **//
+
+// A couple of utilities first. This is just standard HTML entity escaping.
+var escapeHtml = function(s) {
+	var tagsToReplace = {
+		'&': '&amp;',
+		'<': '&lt;',
+		'>': '&gt;'
+	};
+	return s.replace(/[&<>]/g, function(tag) {
+		return tagsToReplace[tag] || tag;
+	});
+};
+
+// https://github.com/domenic/last tweaked slightly so that it works with
+// jQuery's broken promises (we don't want to paste an entire full-size
+// promise library into one of these scripts).
+var last = (function() {
+	// 'pending' is a promise that never resolves - it's returned from
+	// then() below to make other promises also never resolve.
+	var pending = $.Deferred().promise();
+	return function(operation) {
+		var latestPromise = null;
+		return function() {
+			var promiseForResult = operation.apply(this, arguments);
+			latestPromise = promiseForResult;
+
+			return promiseForResult.then(
+				function(value) {
+					if (latestPromise === promiseForResult) {
+						return value;
+					} else {
+						return pending;
+					}
+				}, function(reason) {
+					if (latestPromise === promiseForResult) {
+						throw reason;
+					} else {
+						return pending;
+					}
+				}
+			);
+		};
+	};
+})();
+
+// Most of the AJAX calls we use take multipart form data as their content,
+// so this helper makes it a little more convenient to send that sort of
+// thing up.
+var postFormData = function(url, data) {
+	var formData = new FormData(), k;
+	for (k in data) if (data.hasOwnProperty(k)) {
+		formData.append(k, data[k]);
+	}
+	return $.ajax({
+		url: url,
+		type: 'POST',
+		data: formData,
+		headers: {'x-tumblr-form-key': XKit.interface.form_key()},
+		processData: false,
+		contentType: false
+	});
+};
+
+var ACE_SELECTOR = '.html-field .ace_editor';
+var aceInsert = function(str) {
+	XKit.tools.unsafe_eval(function() {
```

Comment:
I believe that this can use the existing `XKit.tools.add_function` as defined in https://github.com/new-xkit/XKit/blob/master/Extensions/xkit_patches.js#L215